internal/controller/pkg/revision/deployment_test.go

```diff
@@ -0,0 +1,159 @@
+package revision
```

Comment:
Thank you for adding these tests @chlunde!

Reply:
Lesson learned, even for a "trivial" change, add tests even if the file does not have any already :)

PS! This doesn't add full coverage for all the existing branches here